RadixWallet/Features/SplashFeature/Splash.swift

```diff
@@ -72,7 +80,21 @@ public struct Splash: Sendable, FeatureReducer {
 	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
 		switch viewAction {
 		case .appeared:
-			return delay().concatenate(with: verifyPasscode())
+			return .run { send in
+				let isAdvancedLockEnabled = await onboardingClient.loadProfile().appPreferences.security.isAdvancedLockEnabled
+
+				// Starting with iOS 18, the system-provided biometric check will be used
+				if #unavailable(iOS 18), isAdvancedLockEnabled {
+					#if targetEnvironment(simulator)
+					let action: InternalAction = _XCTIsTesting ? .checkBiometrics : .loadingDataCompleted
+					#else
+					let action: InternalAction = .checkBiometrics
+					#endif
+					await send(.internal(action))
+				} else {
+					await send(.internal(.loadingDataCompleted))
+				}
+			}
```

Comment:
I would suggest to introduce a single internal action, such as `advancedLockStateLoaded(isEnabled: Bool)`, which sent from this run block.
Then you could remove the other two actions `checkBiometrics` and `loadingDataCompleted`, in favour of having:
```swift
case .let advancedLockStateLoaded(isEnabled):
   return isEnabled ? verifyPasscode() : delegateCompleted()
```